ush/setup.py

```diff
@@ -1921,24 +1282,21 @@ def get_location(xcs, fmt):
     # -----------------------------------------------------------------------
     #
 
-    # loop through cfg_d and check validity of params
+    # loop through the flattened expt_config and check validity of params
     cfg_v = load_config_file("valid_param_vals.yaml")
-    cfg_d = flatten_dict(cfg_d)
-    for k, v in cfg_d.items():
-        if v == None:
+    for k, v in flatten_dict(expt_config).items():
+        if v is None or v == '':
             continue
         vkey = "valid_vals_" + k
         if (vkey in cfg_v) and not (v in cfg_v[vkey]):
             raise Exception(
                 f"""
-                The variable {k}={v} in {EXPT_DEFAULT_CONFIG_FN} or {EXPT_CONFIG_FN}
+                The variable {k}={v} in the user's configuration
                 does not have a valid value. Possible values are:
                     {k} = {cfg_v[vkey]}"""
             )
 
-    # add LOGDIR and return flat dict
-    cfg_d.update({"LOGDIR": LOGDIR})
-    return cfg_d
```

Comment:
@danielabdi-noaa I was just taking a look at how this is handled. Do you think it could be an appropriate solution to set LOGBASEDIR in the config_default.yaml and var_defns.sh to avoid the namespace collision at run time?

Reply:
@christinaholtNOAA Yes, i think that will work! The problem is we have two log directories in NCO mode, one under OPSROOT and another in EXPTDIR for symlinks. So if we can use a separate variable for the latter, we can safely store it in var_defns.